bug with DQN

```console
$ sequoia_sweep --setting rl  --max_runs 100  --dataset monsterkong  --nb_task 10  --method dqn  --project monsterkong_sweep  --steps_per_task 200000  --monitor_training_performance
 
2021-03-16:00:57:19,992 INFO     [sequoia/settings/assumptions/incremental.py:232] Finished Training on task 0.

Test:   0%|          | 0/10000 [00:00<?, ?it/s]
Test:   0%|          | 8/10000 [00:00<00:45, 219.72it/s]
Traceback (most recent call last):
  File "/mnt/home/dev/Sequoia/sequoia/settings/assumptions/incremental.py", line 382, in test_loop
    method.test(test_env)
  File "/mnt/home/dev/Sequoia/sequoia/settings/base/bases.py", line 307, in test
    raise NotImplementedError
NotImplementedError
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/mnt/home/.conda/envs/sequoia/bin/sequoia_sweep", line 33, in <module>
    sys.exit(load_entry_point('sequoia', 'console_scripts', 'sequoia_sweep')())
  File "/mnt/home/dev/Sequoia/sequoia/experiments/hpo_sweep.py", line 122, in main
    return HPOSweep.main()
  File "/mnt/home/dev/Sequoia/sequoia/experiments/hpo_sweep.py", line 118, in main
    return experiment.launch(argv, strict_args=strict_args)
  File "/mnt/home/dev/Sequoia/sequoia/experiments/hpo_sweep.py", line 68, in launch
    best_params, best_objective = self.method.hparam_sweep(
  File "/mnt/home/dev/Sequoia/sequoia/settings/base/bases.py", line 706, in hparam_sweep
    result: Results = setting.apply(self)
  File "/mnt/home/dev/Sequoia/sequoia/settings/active/continual/continual_rl_setting.py", line 565, in apply
    results = self.main_loop(method)
  File "/mnt/home/dev/Sequoia/sequoia/settings/assumptions/incremental.py", line 233, in main_loop
    test_metrics: TaskSequenceResults = self.test_loop(method)
  File "/mnt/home/dev/Sequoia/sequoia/settings/assumptions/incremental.py", line 426, in test_loop
    action = method.get_actions(obs, action_space)
  File "/mnt/home/dev/Sequoia/sequoia/methods/stable_baselines3_methods/dqn.py", line 175, in get_actions
    return super().get_actions(
  File "/mnt/home/dev/Sequoia/sequoia/methods/stable_baselines3_methods/base.py", line 279, in get_actions
    assert action in action_space, (observations, action, action_space)
AssertionError: (ContinualRLSetting.Observations(x=array([[[160,  97,  40],
        [202, 124,  51],
        [183, 112,  47],
        ...,
        [190, 117,  50],
        [203, 124,  52],
        [166,  98,  42]],
       [[189, 117,  47],
        [211, 130,  53],
        [191, 117,  47],
        ...,
        [216, 137,  54],
        [209, 129,  52],
        [178, 106,  44]],
       [[170, 104,  41],
        [195, 120,  49],
        [174, 107,  44],
        ...,
        [201, 125,  51],
        [191, 117,  48],
        [166,  98,  42]],
       ...,
       [[178, 112,  45],
        [217, 138,  55],
        [196, 124,  51],
        ...,
        [209, 134,  55],
        [218, 138,  55],
        [181, 110,  47]],
       [[187, 116,  46],
        [211, 130,  52],
        [189, 116,  47],
        ...,
        [214, 136,  54],
        [210, 129,  52],
        [177, 106,  44]],
       [[156,  93,  37],
        [182, 110,  46],
        [166,  99,  42],
        ...,
        [186, 112,  47],
        [178, 108,  45],
        [160,  92,  40]]], dtype=uint8), task_labels=0, done=None), array([4, 4, 4, 4, 2, 0, 3, 4, 4, 1, 3, 4, 5, 3, 1, 5, 4, 0, 1, 5, 3, 4,
       4, 2, 5, 5, 0, 4, 3, 2, 5, 5, 2, 4, 4, 5, 2, 1, 0, 3, 4, 0, 0, 2,
       0, 1, 3, 0, 5, 5, 4, 4, 3, 4, 3, 4, 4, 1, 5, 0, 2, 5, 1, 2]), Discrete(6))
wandb: Network error (ReadTimeout), entering retry loop. See wandb/debug-internal.log for full traceback.
wandb: Network error resolved after 0:00:58.176990, resuming normal operation.
wandb: Network error (HTTPError), entering retry loop. See wandb/debug-internal.log for full traceback.
wandb: Network error resolved after 0:01:26.480207, resuming normal operation.
wandb: Network error (HTTPError), entering retry loop. See wandb/debug-internal.log for full traceback.
wandb: Network error resolved after 0:00:08.483608, resuming normal operation.
wandb: Network error (HTTPError), entering retry loop. See wandb/debug-internal.l
```

--- a/examples/clcomp21/classifier_test.py
+++ b/examples/clcomp21/classifier_test.py
@@ -15,7 +15,7 @@ def test_mnist(mnist_setting: SettingProxy[ClassIncrementalSetting]):
     assert results.to_log_dict()
 
     results: ClassIncrementalSetting.Results
-    assert 0.80 <= results.average_online_performance.objective <= 1.00
+    assert 0.60 <= results.average_online_performance.objective <= 1.00
     assert 0.10 <= results.average_final_performance.objective <= 0.30
 
 

--- a/examples/clcomp21/multihead_classifier_test.py
+++ b/examples/clcomp21/multihead_classifier_test.py
@@ -19,7 +19,7 @@ def test_mnist(mnist_setting: SettingProxy[ClassIncrementalSetting]):
     results: ClassIncrementalSetting.Results
     # There should be an improvement over the Method in `classifier.py`:
     assert 0.80 <= results.average_online_performance.objective <= 1.00
-    assert 0.20 <= results.average_final_performance.objective <= 0.50
+    assert 0.20 <= results.average_final_performance.objective <= 0.60
 
 
 @slow

--- a/sequoia/common/batch.py
+++ b/sequoia/common/batch.py
@@ -134,9 +134,10 @@ class Batch(ABC, Mapping[str, T]):
     """
     # TODO: Would it make sense to add a gym Space class variable here? 
     space: ClassVar[Optional[gym.Space]]
+    # TODO: Remove these:
     field_names: ClassVar[List[str]]
     _namedtuple: ClassVar[Type[NamedTuple]]
-    
+
     def __init_subclass__(cls, *args, **kwargs):
         # IDEA: By not marking 'Batch' a dataclass, we would let the subclass
         # decide it if wants to be frozen or not!

--- a/sequoia/common/gym_wrappers/transform_wrappers_test.py
+++ b/sequoia/common/gym_wrappers/transform_wrappers_test.py
@@ -1,0 +1,22 @@
+import gym
+import numpy as np
+from sequoia.common.spaces import Image
+from sequoia.common.transforms import Compose, Transforms
+from sequoia.conftest import monsterkong_required
+
+from .transform_wrappers import TransformObservation
+
+
+@monsterkong_required
+def test_compose_on_image_space():
+    in_space = Image(0, 255, shape=(64, 64, 3), dtype=np.uint8)
+    transform = Compose([Transforms.to_tensor, Transforms.three_channels])
+    expected = Image(0, 1., shape=(3, 64, 64), dtype=np.float32) 
+    actual = transform(in_space)
+   
+    assert actual == expected
+    env = gym.make("MetaMonsterKong-v0")
+    assert env.observation_space == gym.spaces.Box(0, 255, (64, 64, 3), np.uint8)
+    assert env.observation_space == in_space
+    wrapped_env = TransformObservation(env, transform)
+    assert wrapped_env.observation_space == expected

--- a/sequoia/methods/stable_baselines3_methods/a2c_test.py
+++ b/sequoia/methods/stable_baselines3_methods/a2c_test.py
@@ -1,0 +1,42 @@
+import pytest
+from sequoia.common.config import Config
+from sequoia.conftest import monsterkong_required
+from sequoia.settings.active import (
+    ContinualRLSetting,
+    IncrementalRLSetting,
+    TaskIncrementalRLSetting,
+)
+
+from .a2c import A2CMethod
+
+
+def test_cartpole_state():
+    method = A2CMethod()
+    setting = IncrementalRLSetting(
+        dataset="cartpole",
+        observe_state_directly=True,
+        nb_tasks=2,
+        steps_per_task=1_000,
+        test_steps_per_task=1_000,
+    )
+    results: IncrementalRLSetting.Results = setting.apply(
+        method, config=Config(debug=True)
+    )
+    print(results.summary())
+
+
+@pytest.mark.timeout(120)
+@monsterkong_required
+def test_monsterkong():
+    method = A2CMethod()
+    setting = IncrementalRLSetting(
+        dataset="monsterkong",
+        nb_tasks=2,
+        steps_per_task=1_000,
+        test_steps_per_task=1_000,
+    )
+    results: IncrementalRLSetting.Results = setting.apply(
+        method, config=Config(debug=True)
+    )
+    print(results.summary())
+

--- a/sequoia/methods/stable_baselines3_methods/base.py
+++ b/sequoia/methods/stable_baselines3_methods/base.py
@@ -217,10 +217,18 @@ class StableBaselines3Method(Method, ABC, target_setting=ContinualRLSetting):
         # BUG: Need to fix an issue when using the CnnPolicy and Atary envs, the
         # input shape isn't what they expect (only 2 channels instead of three
         # apparently.)
-        setting.transforms = []
-        setting.train_transforms = []
-        setting.val_transforms = []
-        setting.test_transforms = []
+        from sequoia.common.transforms import Transforms
+        # NOTE: Important to not use any transforms, since the SB3 methods want to get
+        # the 'raw' np.uint8 image as an input.
+        transforms = [
+            # Transforms.to_tensor,
+            # Transforms.three_channels,
+            # Transforms.channels_first_if_needed,
+        ]
+        setting.transforms = transforms
+        setting.train_transforms = transforms
+        setting.val_transforms = transforms
+        setting.test_transforms = transforms
 
         if self.hparams.policy is None:
             if setting.observe_state_directly:

--- a/sequoia/methods/stable_baselines3_methods/ddpg_test.py
+++ b/sequoia/methods/stable_baselines3_methods/ddpg_test.py
@@ -1,0 +1,43 @@
+import pytest
+from sequoia.common.config import Config
+from sequoia.conftest import monsterkong_required
+from sequoia.settings.active import (
+    ContinualRLSetting,
+    IncrementalRLSetting,
+    TaskIncrementalRLSetting,
+)
+
+from .ddpg import DDPGMethod
+
+
+@pytest.mark.xfail(reason="DDPG needs continuous action spaces.")
+def test_cartpole_state():
+    method = DDPGMethod()
+    setting = IncrementalRLSetting(
+        dataset="cartpole",
+        observe_state_directly=True,
+        nb_tasks=2,
+        steps_per_task=1_000,
+        test_steps_per_task=1_000,
+    )
+    results: IncrementalRLSetting.Results = setting.apply(
+        method, config=Config(debug=True)
+    )
+    print(results.summary())
+
+
+@pytest.mark.xfail(reason="DDPG needs continuous action spaces.")
+@monsterkong_required
+def test_monsterkong():
+    method = DDPGMethod()
+    setting = IncrementalRLSetting(
+        dataset="monsterkong",
+        nb_tasks=2,
+        steps_per_task=1_000,
+        test_steps_per_task=1_000,
+    )
+    results: IncrementalRLSetting.Results = setting.apply(
+        method, config=Config(debug=True)
+    )
+    print(results.summary())
+

--- a/sequoia/methods/stable_baselines3_methods/dqn.py
+++ b/sequoia/methods/stable_baselines3_methods/dqn.py
@@ -11,24 +11,27 @@ from gym.spaces.utils import flatten_space
 from simple_parsing import mutable_field
 from stable_baselines3.dqn import DQN
 
-from sequoia.common.hparams import uniform, log_uniform, categorical
+from sequoia.common.hparams import categorical, log_uniform, uniform
+from sequoia.common.transforms import ChannelsFirst
 from sequoia.methods import register_method
-from sequoia.methods.stable_baselines3_methods.base import (
-    SB3BaseHParams, StableBaselines3Method)
 from sequoia.settings.active import ContinualRLSetting
 from sequoia.utils.logging_utils import get_logger
+
+from .base import SB3BaseHParams, StableBaselines3Method
 
 logger = get_logger(__file__)
 
 
 class DQNModel(DQN):
     """ Customized version of the DQN model from stable-baselines-3. """
+
     @dataclass
     class HParams(SB3BaseHParams):
         """ Hyper-parameters of the DQN model from `stable_baselines3`.
 
         The command-line arguments for these are created with simple-parsing.
         """
+
         # The learning rate, it can be a function of the current progress (from
         # 1 to 0)
         learning_rate: Union[float, Callable] = log_uniform(1e-6, 1e-2, default=1e-4)
@@ -169,13 +172,17 @@ class DQNMethod(StableBaselines3Method):
     def fit(self, train_env: gym.Env, valid_env: gym.Env):
         super().fit(train_env=train_env, valid_env=valid_env)
 
-    def get_actions(self,
-                    observations: ContinualRLSetting.Observations,
-                    action_space: spaces.Space) -> ContinualRLSetting.Actions:
-        return super().get_actions(
-            observations=observations,
-            action_space=action_space,
-        )
+    def get_actions(
+        self, observations: ContinualRLSetting.Observations, action_space: spaces.Space
+    ) -> ContinualRLSetting.Actions:
+        obs = observations.x
+        # Temp fix for monsterkong and DQN:
+        if obs.shape == (64, 64, 3):
+            obs = ChannelsFirst.apply(obs)
+        predictions = self.model.predict(obs)
+        action, _ = predictions
+        assert action in action_space, (observations, action, action_space)
+        return action
 
     def on_task_switch(self, task_id: Optional[int]) -> None:
         """ Called when switching tasks in a CL setting.

--- a/sequoia/methods/stable_baselines3_methods/dqn_test.py
+++ b/sequoia/methods/stable_baselines3_methods/dqn_test.py
@@ -1,0 +1,96 @@
+import numpy as np
+import pytest
+import torch
+from gym import spaces
+from sequoia.common.config import Config
+from sequoia.common.spaces import Image, NamedTupleSpace, Sparse
+from sequoia.conftest import monsterkong_required
+from sequoia.settings.active import (
+    ContinualRLSetting,
+    IncrementalRLSetting,
+    TaskIncrementalRLSetting,
+)
+
+from .dqn import DQNMethod
+
+
+def test_cartpole_state():
+    method = DQNMethod()
+    setting = IncrementalRLSetting(
+        dataset="cartpole",
+        observe_state_directly=True,
+        nb_tasks=2,
+        steps_per_task=1_000,
+        test_steps_per_task=1_000,
+    )
+    results: IncrementalRLSetting.Results = setting.apply(
+        method, config=Config(debug=True)
+    )
+    print(results.summary())
+
+
+@monsterkong_required
+def test_monsterkong():
+    method = DQNMethod()
+    setting = IncrementalRLSetting(
+        dataset="monsterkong",
+        nb_tasks=2,
+        steps_per_task=1_000,
+        test_steps_per_task=1_000,
+    )
+    results: IncrementalRLSetting.Results = setting.apply(
+        method, config=Config(debug=True)
+    )
+    print(results.summary())
+
+
+def test_dqn_monsterkong_adds_channel_first_transform():
+    method = DQNMethod()
+    setting = IncrementalRLSetting(
+        dataset="monsterkong",
+        nb_tasks=2,
+        steps_per_task=1_000,
+        test_steps_per_task=1_000,
+    )
+    assert setting.max_steps == 2_000
+    assert setting.test_steps == 2_000
+    assert setting.nb_tasks == 2
+    assert setting.observation_space == NamedTupleSpace(
+        spaces={
+            "x": Image(0, 1, shape=(3, 64, 64), dtype=np.float32),
+            "task_labels": Sparse(spaces.Discrete(2)),
+        },
+        dtype=setting.Observations,
+    )
+    assert setting.action_space == spaces.Discrete(6)  # monsterkong has 6 actions.
+
+    # (Before the method gets to change the Setting):
+    # By default the setting gives the same shape of obs as the underlying env.
+    for env_method in [
+        setting.train_dataloader,
+        setting.val_dataloader,
+        setting.test_dataloader,
+    ]:
+        print(f"Testing method {env_method.__name__}")
+        with env_method() as env:
+            reset_obs = env.reset()
+            # TODO: Fix this so the 'x' space actually gets tensor support.
+            # assert reset_obs in env.observation_space
+            assert reset_obs.numpy() in env.observation_space
+            assert reset_obs.x.shape == (3, 64, 64)
+
+    # Let the Method configure itself on the Setting:
+    method.configure(setting)
+
+    # (After the method gets to change the Setting):
+
+    for env_method in [
+        setting.train_dataloader,
+        setting.val_dataloader,
+        setting.test_dataloader,
+    ]:
+        with env_method() as env:
+            reset_obs = env.reset()
+            # Fix this numpy bug.
+            assert reset_obs.numpy() in env.observation_space
+            assert reset_obs.x.shape == (64, 64, 3)

--- a/sequoia/methods/stable_baselines3_methods/ppo_test.py
+++ b/sequoia/methods/stable_baselines3_methods/ppo_test.py
@@ -1,0 +1,42 @@
+import pytest
+from sequoia.common.config import Config
+from sequoia.conftest import monsterkong_required
+from sequoia.settings.active import (
+    ContinualRLSetting,
+    IncrementalRLSetting,
+    TaskIncrementalRLSetting,
+)
+
+from .ppo import PPOMethod
+
+
+def test_cartpole_state():
+    method = PPOMethod()
+    setting = IncrementalRLSetting(
+        dataset="cartpole",
+        observe_state_directly=True,
+        nb_tasks=2,
+        steps_per_task=1_000,
+        test_steps_per_task=1_000,
+    )
+    results: IncrementalRLSetting.Results = setting.apply(
+        method, config=Config(debug=True)
+    )
+    print(results.summary())
+
+
+@pytest.mark.timeout(120)
+@monsterkong_required
+def test_monsterkong():
+    method = PPOMethod()
+    setting = IncrementalRLSetting(
+        dataset="monsterkong",
+        nb_tasks=2,
+        steps_per_task=1_000,
+        test_steps_per_task=1_000,
+    )
+    results: IncrementalRLSetting.Results = setting.apply(
+        method, config=Config(debug=True)
+    )
+    print(results.summary())
+

--- a/sequoia/methods/stable_baselines3_methods/sac_test.py
+++ b/sequoia/methods/stable_baselines3_methods/sac_test.py
@@ -1,0 +1,45 @@
+import pytest
+from sequoia.common.config import Config
+from sequoia.conftest import monsterkong_required
+from sequoia.settings.active import (
+    ContinualRLSetting,
+    IncrementalRLSetting,
+    TaskIncrementalRLSetting,
+)
+
+
+from .sac import SACMethod
+import pytest
+
+
+@pytest.mark.xfail(reason="needs continuous action space")
+def test_cartpole_state():
+    method = SACMethod()
+    setting = IncrementalRLSetting(
+        dataset="cartpole",
+        observe_state_directly=True,
+        nb_tasks=2,
+        steps_per_task=1_000,
+        test_steps_per_task=1_000,
+    )
+    results: IncrementalRLSetting.Results = setting.apply(
+        method, config=Config(debug=True)
+    )
+    print(results.summary())
+
+
+@pytest.mark.xfail(reason="needs continuous action space")
+@monsterkong_required
+def test_monsterkong():
+    method = SACMethod()
+    setting = IncrementalRLSetting(
+        dataset="monsterkong",
+        nb_tasks=2,
+        steps_per_task=1_000,
+        test_steps_per_task=1_000,
+    )
+    results: IncrementalRLSetting.Results = setting.apply(
+        method, config=Config(debug=True)
+    )
+    print(results.summary())
+

--- a/sequoia/methods/stable_baselines3_methods/td3_test.py
+++ b/sequoia/methods/stable_baselines3_methods/td3_test.py
@@ -1,0 +1,43 @@
+import pytest
+from sequoia.common.config import Config
+from sequoia.conftest import monsterkong_required
+from sequoia.settings.active import (
+    ContinualRLSetting,
+    IncrementalRLSetting,
+    TaskIncrementalRLSetting,
+)
+
+from .td3 import TD3Method
+
+
+@pytest.mark.xfail(reason="needs continuous action space.")
+def test_cartpole_state():
+    method = TD3Method()
+    setting = IncrementalRLSetting(
+        dataset="cartpole",
+        observe_state_directly=True,
+        nb_tasks=2,
+        steps_per_task=1_000,
+        test_steps_per_task=1_000,
+    )
+    results: IncrementalRLSetting.Results = setting.apply(
+        method, config=Config(debug=True)
+    )
+    print(results.summary())
+
+
+@pytest.mark.xfail(reason="needs continuous action spaces.")
+@monsterkong_required
+def test_monsterkong():
+    method = TD3Method()
+    setting = IncrementalRLSetting(
+        dataset="monsterkong",
+        nb_tasks=2,
+        steps_per_task=1_000,
+        test_steps_per_task=1_000,
+    )
+    results: IncrementalRLSetting.Results = setting.apply(
+        method, config=Config(debug=True)
+    )
+    print(results.summary())
+

--- a/sequoia/settings/active/continual/continual_rl_setting.py
+++ b/sequoia/settings/active/continual/continual_rl_setting.py
@@ -976,8 +976,8 @@ class ContinualRLSetting(ActiveSetting, IncrementalSetting):
         max_steps: int,
         new_random_task_on_reset: bool,
     ) -> List[Callable[[gym.Env], gym.Env]]:
-        """ helper function for creating the train/valid/test wrappers. 
-        
+        """ helper function for creating the train/valid/test wrappers.
+
         These wrappers get applied *before* the batching, if applicable.
         """
         wrappers: List[Callable[[gym.Env], gym.Env]] = []
@@ -1002,12 +1002,14 @@ class ContinualRLSetting(ActiveSetting, IncrementalSetting):
 
         if (
             isinstance(self.dataset, str)
-            and self.dataset.lower().startswith("metamonsterkong")
+            and self.dataset.lower().startswith(("metamonsterkong", "monsterkong"))
             and not self.observe_state_directly
         ):
             # TODO: Do we need the AtariPreprocessing wrapper on MonsterKong?
             # wrappers.append(partial(AtariPreprocessing, frame_skip=1))
+            wrappers.append(ImageObservations)
             pass
+        
         elif is_atari_env(self.dataset):
             # TODO: Test & Debug this: Adding the Atari preprocessing wrapper.
             # TODO: Figure out the differences (if there are any) between the

--- a/sequoia/settings/active/continual/continual_rl_setting.py
+++ b/sequoia/settings/active/continual/continual_rl_setting.py
@@ -464,8 +464,18 @@ class ContinualRLSetting(ActiveSetting, IncrementalSetting):
                     for i, task in enumerate(valid_tasks)
                 }
 
+            space_dict = dict(temp_env.observation_space.items())
+            task_label_space = spaces.Discrete(self.nb_tasks)
+            if not self.task_labels_at_train_time or not self.task_labels_at_test_time:
+                task_label_space = Sparse(task_label_space)
+            space_dict["task_labels"] = task_label_space
+
+            # FIXME: Temporarily, we will actually set the task label space, since there
+            # appears to be an error when using monsterkong space.
+            observation_space = NamedTupleSpace(spaces=space_dict, dtype=self.Observations)
+            self.observation_space = observation_space
             # Set the spaces using the temp env.
-            self.observation_space = temp_env.observation_space
+            # self.observation_space = temp_env.observation_space
             self.action_space = temp_env.action_space
             self.reward_range = temp_env.reward_range
             self.reward_space = getattr(

--- a/sequoia/settings/active/continual/incremental/incremental_rl_setting.py
+++ b/sequoia/settings/active/continual/incremental/incremental_rl_setting.py
@@ -62,7 +62,7 @@ class IncrementalRLSetting(ContinualRLSetting):
             # TODO: Actually end episodes when reaching a task boundary, to force the
             # level to change?
             self.max_episode_steps = self.max_episode_steps or 500
-    
+
     @property
     def phases(self) -> int:
         """The number of training 'phases', i.e. how many times `method.fit` will be

--- a/sequoia/settings/passive/cl/domain_incremental/domain_incremental_setting_test.py
+++ b/sequoia/settings/passive/cl/domain_incremental/domain_incremental_setting_test.py
@@ -32,10 +32,8 @@ def test_domain_incremental_mnist_setup():
             x = x.permute(0, 2, 3, 1)[0]
             assert x.shape == (28, 28, 3)
 
-            reward = train_loader.send([4 for _ in range(batch_size)])
-            # TODO: Why are we fine with getting `None` as the reward here? Is it
-            # because we're somehow setting it to be ``
-            assert reward is None
+            rewards_ = train_loader.send([4 for _ in range(batch_size)])
+            assert (rewards.y == rewards_.y).all()
 
         train_loader.close()
 

--- a/sequoia/settings/passive/cl/task_incremental/task_incremental_setting_test.py
+++ b/sequoia/settings/passive/cl/task_incremental/task_incremental_setting_test.py
@@ -42,7 +42,10 @@ def check_only_right_classes_present(setting: TaskIncrementalSetting):
             assert x.shape == (28, 28, 3)
             
             reward = train_loader.send([4 for _ in range(batch_size)])
-            assert reward is None
+            if setting.monitor_training_performance:
+                assert reward is None
+            else:
+                assert (reward.y == rewards.y).all()
 
         train_loader.close()
 

--- a/tests/examples/quick_demo_test.py
+++ b/tests/examples/quick_demo_test.py
@@ -45,7 +45,7 @@ def test_quick_demo(monkeypatch):
     assert results.final_performance_metrics[4].n_samples == 1984
 
     assert 0.48 <= results.final_performance_metrics[0].accuracy <= 0.55
-    assert 0.48 <= results.final_performance_metrics[1].accuracy <= 0.60
+    assert 0.48 <= results.final_performance_metrics[1].accuracy <= 0.70
     assert 0.60 <= results.final_performance_metrics[2].accuracy <= 0.95
     assert 0.75 <= results.final_performance_metrics[3].accuracy <= 1.00
     assert 0.99 <= results.final_performance_metrics[4].accuracy <= 1.00

